internal/core/docker_host.go

```diff
@@ -116,14 +133,20 @@ func extractDockerHost(ctx context.Context) string {
 			continue
 		}
 
+		err = defaultCallbackCheckFn(ctx, dockerHost)
+		if err != nil {
+			outerErr = fmt.Errorf("%w: %w", outerErr, err)
```

Comment:
nit: I would eliminate the assignment to outerErr as if this is for debug we can do that without it, given its not used.
```suggestion
		if err = defaultCallbackCheckFn(ctx, dockerHost); err != nil {
```

Reply:
Oh good catch! I dragged that from the previous state.

About the outerErr, wdyt on https://github.com/testcontainers/testcontainers-go/pull/2741#discussion_r1727202446?

Reply:
I think we already addressed this. Resolving.